javascript/javascript3/week2/exercise 4/PeopleNames/index.js

```diff
@@ -0,0 +1,29 @@
+function DogNameList() {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve();
+        }, 3000)
+    })}
+   
+   DogNameList()
+   
+   .then(() => {
+       const URL = "https://ghibliapi.herokuapp.com/people";
+       const list = document.getElementById("myList");
+       list.innerHTML = "<p>Loading...";
+       fetch(URL)
+         .then((response) => response.json())
+         .then((people) => list.innerHTML = getListOfNames(people));
+       const getListOfNames = (people) => {
+         const names = people
+           .map((person) => `<li>${person.name}</li>`)
+           .join("\n");
+           console.log(people);
+       return `<ul>${names}</ul>`;
+       };    
```

Comment:
again you put the logic into the `then` rather than into the function `getDogsNames` as with the getLocation example.

instead of nesting the fetch inside the then, could you create a timeoutPromise(), getDogsNames() and chain them like this:

```
timeoutPromise(3000)
.then(getDogsNames)
.then(res => {
    // write to HTML here
})
```

does that make sense?